crates/core_arch/src/x86/avx2.rs

```diff
@@ -191,7 +191,7 @@ pub fn _mm256_alignr_epi8<const IMM8: i32>(a: __m256i, b: __m256i) -> __m256i {
             return transmute(a);
         }
 
-        let r: i8x32 = match IMM8 % 16 {
+        let r: i8x32 = match IMM8 {
```

Comment:
You have to also add a value to the default arm. It is currently `unreachable_unchecked()`, which makes `IMM8` values > 15 UB.